feature/DM-4962-opcua-gateway-java17 (#3812)
SoftwareAG/c8y-docs: content/device-integration/opcua-bundle/upgrading.md

---
weight: 200
title: OPC UA major version upgrade notes
layout: redirect
---

> ### Upgrading from 2021 to 2022 gateway version {#upgrade-2021-to-2022}

Reviewer: ```suggestion
### Upgrading from 1021 to 1022 gateway version {#upgrade-1021-to-1022}
```

> Major version 2022 of the OPC UA gateway introduces significant changes, including a new internal database and updated Java version. 

Reviewer: ```suggestion
Major version 1022 of the OPC UA gateway introduces significant changes, including a new internal database and updated Java version. 
```

> To ensure a smooth transition, please follow these steps:

Reviewer 2: ```suggestion
To ensure a smooth transition, pfollow these steps:
```

> 1. **Backup Existing Data**: Before proceeding with the upgrade, back up your existing gateway data folder to prevent any potential data loss.

Reviewer 2: ```suggestion
1. **Back up the existing data**: Before proceeding with the upgrade, back up your existing gateway data folder to prevent any potential data loss.
```

> 2. **Stop existing gateway**: Ensure that the current gateway instance is stopped before starting the upgrade process.

Reviewer 2: ```suggestion
2. **Stop the existing gateway**: Ensure that the current gateway instance is stopped before starting the upgrade process.
```

> 3. **Download and run data migration tool**: A data migration tool is provided to facilitate the transition of your existing data to the new format. 

Reviewer 2: ```suggestion
3. **Download and run the data migration tool**: A data migration tool is provided to facilitate the transition of your existing data to the new format. 
```

> Download the tool from [here](https://resources.cumulocity.com/examples/opc-ua/migration-tool-for-1022-upgrade/) and follow the instructions in the `README.md` file to execute the migration.

Reviewer 2: ```suggestion
Download the tool from [https://resources.cumulocity.com/examples/opc-ua/migration-tool-for-1022-upgrade/](https://resources.cumulocity.com/examples/opc-ua/migration-tool-for-1022-upgrade/) and follow the instructions in the *README.md* file to execute the migration.
```

4. **Install Java 17**: The new gateway version requires Java 17. Ensure that your system has Java 17 installed and configured properly.

> 5. **Update gateway jar file**: Replace the existing gateway jar file with the new version.

Reviewer 2: ```suggestion
5. **Update the gateway JAR file**: Replace the existing gateway JAR file with the new version.
```

6. **Start the new gateway**: After completing the above steps, start the new gateway instance.

> #### Rollback procedure

Reviewer 2: ```suggestion
#### Rollback procedure {#rollback-procedure}
```
